webapp/src/js/controllers/reports.js

```diff
@@ -197,9 +197,10 @@ angular
         });
     };
 
-    var query = function(options) {
-      options = options || {};
+    var query = function(opts) {
+      const options = Object.assign({}, opts);
       options.limit = options.limit || 50;
+      options.hydrateContactNames = 'hydrateContactNames' in options ? !!options.hydrateContactNames : true;
```

Comment:
This is quite hard to read. What about using `_.extend` as below?

I think the ideal solution would be to destructure `opts`, eg:

```js
const query = ({ limit=50, hydrateContactNames=true }={}) => {
```

This is easy to read and self documenting, however it does require working out all the possible options which isn't easy.

Reply:
For destructuring, options actually has many properties and we send it downstream to Search which takes even more properties. So we'd have to destructure and restructure etc. I went with the `extend` option which is a nice clean command to shallow copy and provide defaults.